spec/intro.dd

```diff
@@ -67,6 +67,68 @@ $(OL
 )
 
 
+$(H2 Memory Model)
+
+    $(P A $(I byte) is the fundamental unit of storage. Each byte has 8 bits. Each byte has
+    a unique address. A $(I memory location) is a sequence of one or more bytes exactly large
+    enough to hold a scalar type. Multiple threads can access separate memory locations
+    without interference.
+    )
+
+    $(P Memory locations come in three groups:)
+
+    $(OL
+    $(LI $(I Thread local memory locations) are accessible from only one thread at a time.)
+    $(LI $(I Immutable memory locations) cannot be written to during its lifetime. Immutable
+    memory locations can be read from by multiple threads without synchronization.)
+    $(LI $(I Shared memory locations) is accessible from multiple threads.)
+    )
+
+    $(UNDEFINED_BEHAVIOR Allowing multiple threads to access a thread local memory
+    location results in undefined behavior.)
+
+    $(UNDEFINED_BEHAVIOR Writing to an immutable memory location during its lifetime
+    results in undefined behavior.)
+
+    $(UNDEFINED_BEHAVIOR Unless synchronized, writing to a shared memory location in
+    one thread while other threads read or write to it results in undefined behavior.)
+
+    $(P Execution of a single thread on thread local and immutable memory locations
+    is $(I sequentially consistent). This means the result of the operations
+    is the same as if it were executed in the order the operations appear in the program.
+    )
+
+
+$(H2 Object Model)
+
+    $(P An $(I object) is created when one of the following is executed:
+    )
+
+    $(UL
+    $(LI a definition)
+    $(LI a $(GLINK2 expression, NewExpression))
+    $(LI a temporary is created)
+    $(LI changing which field of a union is active)
+    )
+
+    $(P An object spans a sequential sequence of memory locations which may or may not
+    be contiguous. It exists during construction, its lifetime, and destruction.
+    Each object has a type, which is determined either statically or by runtime
+    type information.
```

Comment:
I'm pretty sure there can be memory locations whose type is not determined either statically or by runtime type information. What about a `cast(void*)new S` where S does not have a destructor? Maybe this could say while every object conceptually has a type, it may not possible to determine it from either static or runtime information.

Reply:
I think in that case the data is "untyped". I anticipate a followon section which delves into object initialization and lifetimes.